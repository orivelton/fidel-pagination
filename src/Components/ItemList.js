import React from 'react';

const ItemList = ({currentItem, handleModal}) => {
  const {
    currency,
    card: { scheme },
    location: { 
      address,
      city,
      countryCode
    }
  } = currentItem;
  
  return (
    <li className="tags has-addons panel-block item-list" onClick={() => handleModal(currentItem)}>
        <span className="tag is-primary is-large is-light">{currency}</span>
        <span className={scheme === 'visa' ? 'tag is-danger is-large is-light' : 'tag is-link is-large is-light'}>{scheme}</span>
        <span className={'tag is-light is-large'}><address>Address: {`${address}, ${city} - ${countryCode}`}</address></span>
    </li>
  )
};

export default ItemList;
