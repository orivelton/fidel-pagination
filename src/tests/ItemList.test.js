import React from 'react';
import { render } from '@testing-library/react';
import ItemList from '../Components/ItemList';

describe('component App', () => {
  test('render component ItemList', () => {
    const props = {
      "currency": "GBP",
      "programId": "2314f371-39b1-4c80-8040-4144ff1bad09",
      "id": "5f68d602-664a-493f-89d4-901a053077cb",
      "card": {
        "id": "61a401b5-5d92-402f-8d86-89dfa15b515b",
        "lastNumbers": "4009",
        "scheme": "visa"
      },
      "location": {
        "address": "Google Street 4"
      }
    };

    const { container } = render(<ItemList currentItem={props} />);
    expect(container).not.toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});
