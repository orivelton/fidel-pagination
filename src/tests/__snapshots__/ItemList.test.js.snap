// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component App render component ItemList 1`] = `
<li
  class="tags has-addons panel-block item-list"
>
  <span
    class="tag is-primary is-large is-light"
  >
    GBP
  </span>
  <span
    class="tag is-danger is-large is-light"
  >
    visa
  </span>
  <span
    class="tag is-light is-large"
  >
    <address>
      Address: 
      Google Street 4, undefined - undefined
    </address>
  </span>
</li>
`;
